File: frontend-X2/lib/modules/intervention/domain/provider_schedule_entry.dart
import '../../../core/utils/date_parser.dart';
import 'intervention.dart';

/// Entree "publique" du planning d'un prestataire (date + heures occupees +
/// statut), sans donnee client, utilisee pour griser les creneaux deja pris
/// quand un client consulte le calendrier de ce prestataire.
class ProviderScheduleEntry {
  final DateTime date;
  final List<int> heures;
  final InterventionStatus statut;

  const ProviderScheduleEntry({required this.date, required this.heures, required this.statut});

  factory ProviderScheduleEntry.fromJson(Map<String, dynamic> json) {
    return ProviderScheduleEntry(
      date: parseFlexibleDate(json['date']) ?? DateTime.now(),
      heures: (json['heures'] as List?)?.map((e) => e as int).toList() ?? [],
      statut: interventionStatusFromJson(json['statut']),
    );
  }
}

File: Backend/src/main/java/com/renkotechnologie/backend/modules/booking/dto/ProviderScheduleEntry.java
package com.renkotechnologie.backend.modules.booking.dto;

import com.renkotechnologie.backend.modules.booking.entity.Intervention;
import java.time.LocalDate;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

/**
 * Entree "publique" du planning d'un prestataire : uniquement la date, les
 * heures occupees et le statut, sans aucune donnee client. Sert a griser les
 * creneaux deja pris lorsqu'un autre client consulte le calendrier de ce
 * prestataire pour reserver.
 */
@Getter
@Builder
@AllArgsConstructor
public class ProviderScheduleEntry {

    private LocalDate date;
    private List<Integer> heures;
    private String statut;

    public static ProviderScheduleEntry fromEntity(Intervention i) {
        return ProviderScheduleEntry.builder()
                .date(i.getDate())
                .heures(i.getHeures())
                .statut(i.getStatut().name())
                .build();
    }
}
